app/controllers/api_keys_controller.rb

```diff
@@ -41,6 +50,9 @@ def update
       flash[:error] = @api_key.errors.full_messages.to_sentence
       render :edit
     end
+  rescue ActiveRecord::RecordNotFound
```

Comment:
Is there a test for this?

Reply:
Yes, should be testing it [here](https://github.com/Shopify/rubygems.org/blob/3f7168f4a527adfafcb500a273d3d36b30e3b349/test/functional/api_keys_controller_test.rb#L226).